### PR TITLE
Unified nuspec

### PR DESCRIPTION
**Pull request by tofutim**
This pull creates a Unified nuspec which uses the Unified dlls instead of Touch and Mac/XamMac.

### CODE REVIEW

**Comment by slodge** [comment]
Wow! :+1: 

Thanks

187 files changed - most of them csproj and nuspec files :)

I've not used unified on Mac yet... I'm guessing from the #914 discussions, that this can only be built on a Mac...

Since this doesn't effect the "main" build, I'm happy to add it - thanks :+1: 

I do have some questions, though, about the  changed .csproj files - several of the Droid and Windows projects now seem to have Unified configurations added - does the Mac build system mean that they now need these?

Sorry I'm such a dunce when it comes to build systems....

Stuart

**Comment by tofutim** [comment]
Dear Stuart,

The Unified configurations was kind of by accident, when I created the
configuration it propagated everywhere. I can clean that up. Currently only
the PCLs and Unified are built by the Unified configuration.

Respectfully,
Tim

**Comment by hollywoof** [comment]
Thanks.

Just reviewing this... 

There's still all sorts of additional configurations in csproj files like:

Cirrious/Cirrious.MvvmCross.AutoView.Droid/Cirrious.MvvmCross.AutoView.Droid.csproj
Cirrious/Cirrious.MvvmCross.AutoView.Droid/Cirrious.MvvmCross.AutoView.Droid.csproj
Cirrious/Cirrious.MvvmCross.Binding.Droid/Cirrious.MvvmCross.Binding.Droid.csproj
Cirrious/Cirrious.MvvmCross.Binding.Touch/Cirrious.MvvmCross.Binding.Touch.csproj
 Cirrious/Cirrious.MvvmCross.Binding/Cirrious.MvvmCross.Binding.csproj
Cirrious/Cirrious.MvvmCross.BindingEx.WindowsPhone/Cirrious.MvvmCross.BindingEx.WindowsPhone.csproj
Cirrious/Cirrious.MvvmCross.BindingEx.WindowsStore/Cirrious.MvvmCross.BindingEx.WindowsStore.csproj
Cirrious/Cirrious.MvvmCross.BindingEx.Wpf/Cirrious.MvvmCross.BindingEx.Wpf.csproj
Cirrious/Cirrious.MvvmCross.Console/Cirrious.MvvmCross.Console.csproj
Cirrious/Cirrious.MvvmCross.Dialog.Droid/Cirrious.MvvmCross.Dialog.Droid.csproj 
Cirrious/Cirrious.MvvmCross.Dialog.Touch/Cirrious.MvvmCross.Dialog.Touch.csproj
Cirrious/Cirrious.MvvmCross.Droid.Fragging/Cirrious.MvvmCross.Droid.Fragging.csproj
Cirrious/Cirrious.MvvmCross.Droid.FullFragging/Cirrious.MvvmCross.Droid.FullFragging.csproj
Cirrious/Cirrious.MvvmCross.Droid.Maps/Cirrious.MvvmCross.Droid.Maps.csproj View
Cirrious/Cirrious.MvvmCross.Droid/Cirrious.MvvmCross.Droid.csproj
OK... I'm stopping - easier to just say "etc..." :)

I guess it's probably easiest for me just to merge in the `mac` projects only?

I'm happy (keen!) to import these changes - just scared that once I merge in extra configurations, then they will never go away again...

Stuart

**Comment by tofutim** [comment]
I have a fix for that coming soon, just hold on. I'll PR tomorrow.

**Comment by tofutim** [comment]
Ok, now let us pull this. I cleaned up all those configurations. Please check.

**Comment by slodge** [comment]
221 changed files!

Top work :+1:

Please give me a day (or two) to find a slot to read these through (and work out what the heck half of these csproj settings mean - lots of those project file changes mean nothing to me!). I'll aim to pull this in and to try to test the droid, touch, winphone and win platforms - that'll take quite a few hours.... If you're checking anything else in on this pull request, then let me know - 221 files is already plenty to try to understand :)

**Comment by tofutim** [comment]
I took out those errant configurations.

**Comment by tofutim** [comment]
(The problem previously was that when you add a new configuration, it asks
if you want to create it in the projects, and unfortunately I said ok. But
I have since stripped them out again.)

File: CrossUI/CrossUI.Touch/Dialog/Elements/HtmlElement.cs [hunk 1]
```diff
@@ -125,10 +125,7 @@ public override void Selected(DialogViewController dvc, UITableView tableView, N
                 {
                     NetworkActivity = false;
                     vc.NavigationItem.RightBarButtonItem = null;
-                    if (_web != null)
-                        _web.LoadHtmlString(
-                            String.Format("<html><center><font size=+5 color='red'>{0}:<br>{1}</font></center></html>",
-                                          "An error occurred:".GetText(), args.Error.LocalizedDescription), null);
+      </PropertyGroup>LocalizedDescription), null);
```
Comment (slodge) [inline CrossUI/CrossUI.Touch/Dialog/Elements/HtmlElement.cs]: This looks wrong? Or have I got something muddled?

File: Plugins/Cirrious/Color/Cirrious.MvvmCross.Plugins.Color.WindowsCommon/Cirrious.MvvmCross.Plugins.Color.WindowsCommon.csproj [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (slodge) [inline Plugins/Cirrious/Color/Cirrious.MvvmCross.Plugins.Color.WindowsCommon/Cirrious.MvvmCross.Plugins.Color.WindowsCommon.csproj]: This ToolsVersion change scares me a bit - but hopefully VS will ignore XS (let's wait and see :))

File: Samples/ApiExamples/ApiExamples.Mac/ApiExamples.Mac.csproj [hunk 3]
```diff
@@ -27,6 +25,7 @@
     <CodeSigningKey>Mac Developer</CodeSigningKey>
     <EnableCodeSigning>false</EnableCodeSigning>
     <CreatePackage>false</CreatePackage>
+    <UseRefCounting>false</UseRefCounting>
```
Comment (slodge) [inline Samples/ApiExamples/ApiExamples.Mac/ApiExamples.Mac.csproj]: Are these really needed? :/

**Comment by slodge** [comment]
Thanks

This looks really cleaned up

Huge effort

Thanks :+1: 

Long (short?) term, i'd quite like to get the samples out of this repo again... it seems like this repo is big enough already even without samples :)

I'm currently waiting on Xam to unlock my licenses before I can build stuff.... but will get this integrated soon....

Stuart

**Comment by tofutim** [comment]
I fine with taking out the samples. Maybe you can give me an idea of how
you develop without having the samples in there. I find I need the samples
to test things out.

**Comment by tofutim** [comment]
Regarding Xamarin, let me see what I can do to help.

**Comment by slodge** [comment]
For samples, I generally use the nuget packages...

Or I use folder sharing across to the /bin folder for includes...

I haven't used direct project references for a long time... not since v2 (maybe)

Others also use git sub repos to the binaries project (but that doesn't include mac binaries today - because it's built on Windows)

File: CrossUI/CrossUI.Touch/Dialog/Elements/HtmlElement.cs (hunk 1 above)
Comment (slodge) [inline CrossUI/CrossUI.Touch/Dialog/Elements/HtmlElement.cs]: Can you check this change - it does look wrong...

**Comment by tofutim** [comment]
Maybe folder sharing is a good idea. Nuget is a bit hard when things keep
changing.

File: CrossUI/CrossUI.Touch/Dialog/Elements/HtmlElement.cs (hunk 1 above)
Comment (tofutim) [inline CrossUI/CrossUI.Touch/Dialog/Elements/HtmlElement.cs]: You can undo that, I don't think I worked on that.

File: Samples/ApiExamples/ApiExamples.Mac/ApiExamples.Mac.csproj (hunk 3 above)
Comment (slodge) [inline Samples/ApiExamples/ApiExamples.Mac/ApiExamples.Mac.csproj]: Sorry - just to be clear the ":/" was me feeling sad if these changes are really needed. I was kind of hoping the new "UseRefCounting" GC would work now after the changes we made to the binding context recently.

**Comment by slodge** [comment]
> Regarding Xamarin, let me see what I can do to help.

Relax - I just did some polite nudging and now I have the update promised a few weeks ago (and a few months ago and...) 

It was polite, honest :)

**Comment by slodge** [comment]
OK

Am pushing 3.5.1-beta1 right now... tests went ok, so just have release notes to do...

Will then branch to get this code, reverse the HtmlElement change and then work out if VS works OK with the changes - hopefully it will :)

More tomorrow morning and evening :)

**Comment by tofutim** [comment]
Stuart, please contact patrick.suzuki@xamarin.com regarding your Xamarin
license. He will facilate the process.

**Comment by tofutim** [comment]
(I'm about to renew, and said I would renew for more years if they could
get you setup quickly.)

**Comment by slodge** [comment]
Slightly slower than planned (work and too much exercise!) but I think this is now there :)

I've tested the non-mac stuff all loads and builds in windows... not tested the mac stuff though... sorry...

:+1: Thanks :)

**Pull request merged**